src/raftstore/store/peer.rs

```diff
@@ -690,26 +679,25 @@ impl Peer {
             });
         }
 
-        ReadyResult {
-            apply_snap_result: apply_snap_result,
-            exec_results: vec![],
+        if let Some(ref res) = apply_snap_result {
+            let snap = ApplyTask::snapshot(self, res.status.clone());
+            self.apply_scheduler.schedule(snap).unwrap();
         }
+
+        apply_snap_result
     }
 
-    pub fn handle_raft_ready_apply(&mut self, mut ready: Ready, result: &mut ReadyResult) {
+    pub fn handle_raft_ready_apply(&mut self, mut ready: Ready) {
         // Call `handle_raft_committed_entries` directly here may lead to inconsistency.
         // In some cases, there will be some pending committed entries when applying a
         // snapshot. If we call `handle_raft_committed_entries` directly, these updates
         // will be written to disk. Because we apply snapshot asynchronously, so these
         // updates will soon be removed. But the soft state of raft is still be updated
         // in memory. Hence when handle ready next time, these updates won't be included
         // in `ready.committed_entries` again, which will lead to inconsistency.
-        result.exec_results = if self.is_applying() {
-            if let Some(ref mut hs) = ready.hs {
-                // Snapshot's metadata has been applied.
-                hs.set_commit(self.get_store().truncated_index());
-            }
-            vec![]
+        if self.is_applying() {
```

Comment:
seem we should use is_applying_snapshot to distinguish with the apply worker doing.